packages/eyeglass/src/assets/Assets.ts

```diff
@@ -1,242 +1,275 @@
-  * @param    {Function} installer - the new installer function
-  */
-Assets.prototype.installer = function(installer) {
-  var oldInstaller = this.install.bind(this);
-  this.install = function(assetFile, assetUri, cb) {
-    installer(assetFile, assetUri, oldInstaller, cb);
-  };
-};
-
-function resolveAssetDefaults(registeredAssetsMap, relativePath) {
-  registeredAssetsMap = this.sassUtils.handleEmptyMap(registeredAssetsMap);
-  this.sassUtils.assertType(registeredAssetsMap, "map");
-
-  var registeredAssets = this.sassUtils.castToJs(registeredAssetsMap);
-
-  var appAssets = registeredAssets.coerce.get(null);
-
-  if (appAssets) {
-    // XXX sassUtils.assertType(appAssets, "map");
-    var appAsset = appAssets.coerce.get(relativePath);
-    if (appAsset) {
-      return appAsset;
+  /**
+    * resolves the asset uri
+    * @param    {String} assetFile - the source file path
+    * @param    {String} assetUri - the resolved uri path
+    * @param    {Function} cb - the callback to pass the resolved uri to
+    */
+  resolve(assetFile: string, assetUri: string, cb: ResolverCallback) {
+    cb(null, { path: assetUri });
+  }
+  /**
+    * wraps the current resolver with a custom resolver
+    * @param    {Function} resolver - the new resolver function
+    */
+  resolver(resolver: WrappedResolver) {
+    let oldResolver: Resolver = this.resolve.bind(this);
+    this.resolve = function(assetFile: string, assetUri: string, cb: ResolverCallback) {
+      resolver(assetFile, assetUri, oldResolver, cb);
+    };
+  }
+  /**
+    * installs the given asset
+    * @param    {String} file - the source file path to install from
+    * @param    {String} uri - the resolved uri path
+    * @param    {Function} cb - the callback invoked after the installation is successful
+    */
+  install(file: string, uri: string, cb: InstallerCallback) {
+    let options = this.eyeglass.options.eyeglass;
+    let httpRoot = options.httpRoot;
+    if (options.buildDir) {
+      // normalize the uri using the system OS path separator
+      // and make it relative to the httpRoot
+      uri = (new URI(uri)).getPath(path.sep, httpRoot);
+
+      let dest = path.join(options.buildDir, uri);
+
+      try {
+        if (options.installWithSymlinks) {
+          fs.mkdirpSync(path.dirname(dest));
+
+          ensureSymlink(file, dest);
+        } else {
+          // we explicitly use copySync rather than copy to avoid starving system resources
+          fs.copySync(file, dest);
+        }
+        cb(null, dest);
+      } catch (error) {
+        // eslint-disable-next-line no-ex-assign
+        error = new Error("Failed to install asset from " + file + "\n" + error.toString());
+
+        cb(error);
+      }
+    } else {
+      cb(null, file);
     }
   }
+  /**
+    * wraps the current installer with a custom installer
+    * @param    {Function} installer - the new installer function
+    */
+  installer(installer: WrappedInstaller) {
```

Comment:
This wrapped caller approach is part of our public API and so I don't think it's a change I can make easily. But it would be good to go over this code with you sometime and see if it's worth refactoring it.